app/models/header_lookup.rb

```diff
@@ -99,6 +105,27 @@ def lcsh_term_pid_lookup(lcsh_term="")
     end
   end
 
+  def lcnaf_pid_lookup(lcnaf_term)
+    lcnaf_term = downcase_and_remove_non_alphanumeric(lcnaf_term)
+
+    puts lcnaf_term
+    # lcnaf file is the trimmmed file here
+    @@searchable_lcnaf_file.each_line do |line|
+      # get the term and id from the file, trim off quotes and newlines
+      term, pid = line.split('":').map{ |val| val.send(:gsub, /[\"\n]/, "") }
+
+      # if the term matches up memoize, write to file, and return pid
+      if lcnaf_term == downcase_and_remove_non_alphanumeric(term)
+        @@memoized_lcnaf[lcnaf_term] = pid
+        File.write(MEMOIZED_LCNAF_FILE, @@memoized_lcnaf)
+        return pid
+      end
+    end
+
+    # if no pid is found, return nil
+    nil
+  end
```

Comment:
When does `@@memoized_lcnaf` come into play? Do we want to check if the `lcnaf_term` is memoized first before checking each line of `@@searchable_lcnaf_file`?

Reply:
it's here: https://github.com/galterlibrary/digital-repository/pull/968/files#diff-90953463eea6f383e933bf550db03f1b8ff5dc3e7a4c544327514c9fcdf8c410R39 (above)